sample/petstore-vertx-rx-server/src/main/java/io/swagger/server/api/MainApiVerticle.java

```diff
@@ -0,0 +1,75 @@
+package io.swagger.server.api;
+
+import java.nio.charset.Charset;
+
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.github.phiz71.vertx.swagger.router.OperationIdServiceIdResolver;
+import com.github.phiz71.vertx.swagger.router.SwaggerRouter;
+
+import io.swagger.models.Swagger;
+import io.swagger.parser.SwaggerParser;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.file.FileSystem;
+import io.vertx.core.json.Json;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.web.Router;
+
+public class MainApiVerticle extends AbstractVerticle {

```

Comment:
I can't help you with DI in Vert.X, but I do agree for the optional generation. And I would add that the fact that this MainVerticle also embed an http server can be annoying.
I will create an issue to remember.